src/ui/dashboardpage.cpp

```diff
@@ -309,13 +311,32 @@ void DashboardPage::draw(App *app, Dashboard *dashboard, Mode mode) noexcept {
                 ImPlotAxisFlags axisFlags = ImPlotAxisFlags_None;
                 axisFlags |= is_horizontal ? (ImPlotAxisFlags_LockMin) : (ImPlotAxisFlags_AutoFit | ImPlotAxisFlags_RangeFit);
 
-                ImPlot::SetupAxis(axis, axisLabel.c_str(), axisFlags);
+                const auto estTextSize = ImGui::CalcTextSize(axisLabel.c_str()).x;
+                if (estTextSize >= a.width) {
+                    static const auto estDotSize = ImGui::CalcTextSize("...").x;
+                    const std::size_t newSize    = a.width / std::max(1.f, estTextSize) * axisLabel.length();
+                    ImPlot::SetupAxis(axis, fmt::format("...{}", axisLabel.substr(axisLabel.length() - newSize, axisLabel.length())).c_str(), axisFlags);
+                } else {
+                    ImPlot::SetupAxis(axis, axisLabel.c_str(), axisFlags);
+                }
                 if (is_horizontal && a.min < a.max) {
                     ImPlot::SetupAxisLimits(axis, a.min, a.max);
                 }
             }
             ImPlot::SetupFinish();
 
+            const auto axisSize = ImPlot::GetPlotSize();
+
+            // compute axis pixel width for H and height for V
+            [&plot] {
+                const ImPlotRect axisLimits = ImPlot::GetPlotLimits(IMPLOT_AUTO, IMPLOT_AUTO);
+                const auto       p0         = ImPlot::PlotToPixels(axisLimits.X.Min, axisLimits.Y.Min);
+                const auto       p1         = ImPlot::PlotToPixels(axisLimits.X.Max, axisLimits.Y.Max);
+                const int        xWidth     = static_cast<int>(std::abs(p1.x - p0.x));
+                const int        yHeight    = static_cast<int>(std::abs(p1.y - p0.y));
+                std::for_each(plot.axes.begin(), plot.axes.end(), [xWidth, yHeight](auto &a) { a.width = (a.axis == Dashboard::Plot::Axis::X) ? xWidth : yHeight; });
```

Comment:
rename `axis.width` to `axis.pixel_length`?

Q: Since the plot size is only available after the plot was rendered, if we want to export screenshots, we will have to render twice if we want to export images in headless mode, right? (Which i think is still reasonable, we should just try not to introduce `iterative` layouting that needs an arbitrary number of passes to converge.)